### PR TITLE
fix: Use pinned supported pnpm version by default

### LINKED ISSUE #11730: FrontendUtils fails to find pnpm if only the newer version is installed
### Minimal reproducible example
1. Download Fusion app from start.vaadin.com for v20.
2. Have only pnpm version 6 available in system's path (in my case, 6.0.1).
3. Run `mvn spring-boot:run`.

### Expected behavior

mvn goal should run without issues, as pnpm 6 is more than the required minimum version of pnpm 5.

### Actual behavior

Error: `java.lang.IllegalStateException: Found too old 'pnpm'. If installed into the project 'node_modules', upgrade 'pnpm' to at least 5.0.0`

### Versions:
    - Vaadin / Flow version: v20
    - Java version: irrelevant
    - OS version: irrelevant
    - IDE (if applicable): Eclipse version: 2019-03 (4.11.0)

### Possible fix
The command used to check for pnpm version, `c:\_node_modules\npx.cmd --yes --quiet pnpm@5 --version` should probably be `c:\_node_modules\npx.cmd --yes --quiet pnpm@5+ --version`. In the current form it produces an empty string when only a newer version is available.

### PR DESCRIPTION
**Pull request by mshabarov**
## Description

Introduces a new configuration parameter `pnpm.global` enabling the globally installed pnpm to be used for frontend packages installation.
By default, this parameter is `false` and the pinned supported version of pnpm tool is used: `npx --yes --quiet --ignore-existing pnpm@5.18.10 install`. No version check needed, because the version is pinned.
Once the `pnpm.global` is set to `true`, then the global pnpm is being looked for via `where/which`. If not found, or it's version is too old, the exception is thrown.

Note: since this change is in the middle between bugfix and enhancement, I propose to include it into `master` only to have a 'trial period' before pick to other branches.

Fixes #11730

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendToolsTest.java [hunk 1]
```diff
@@ -465,38 +481,83 @@ public void forceHomeNode_useHomeNpmFirst() throws Exception {
...
-            throws Exception {
-        createStubNode(false, true, baseDir);
-        createFakePnpm("5.15.1");
-        List<String> pnpmCommand = tools.getSuitablePnpm();
-        Assert.assertEquals("expected pnpm version 5.15.1 accepted", "pnpm@5",
-                pnpmCommand.get(pnpmCommand.size() - 1));
+    public void getSuitablePnpm_tooOldGlobalVersionInstalled_throws() {
+        exceptionRule.expect(IllegalStateException.class);
+        exceptionRule.expectMessage(CoreMatchers.containsString(
+                "Found too old globally installed 'pnpm'. Please upgrade 'pnpm' to at least 5.0.0"));
```
Comment (caalador) [inline flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendToolsTest.java:502]: ExpectedException rule is basically deprecated and using `Assert.assertThrows` is preferred.

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendToolsTest.java [hunk 2]
```diff
@@ -465,38 +481,83 @@ public void forceHomeNode_useHomeNpmFirst() throws Exception {
...
+        tools = new FrontendTools(baseDir, () -> vaadinHomeDir,
+                FrontendTools.DEFAULT_NODE_VERSION, new File(baseDir).toURI(),
+                false, true);
+        try {
+            installGlobalPnpm(OLD_PNPM_VERSION);
+            tools.getSuitablePnpm();
```
Comment (caalador) [inline flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendToolsTest.java:508]:
```suggestion
            IllegalStateException e = Assert.assertThrows(
                IllegalStateException.class, () -> tools.getSuitablePnpm());

            Assert.assertTrue("unexpected content in message '"+e.getMessage()+"'", e.getMessage().contains("Found too old globally installed 'pnpm'. Please upgrade 'pnpm' to at least 5.0.0"));
```

**Review by caalador — CHANGES_REQUESTED** [review]

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendToolsTest.java (hunk 2 above)
Comment (mshabarov) [reply flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendToolsTest.java]: Done

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendToolsTest.java (hunk 1 above)
Comment (mshabarov) [reply flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendToolsTest.java]: Done

**Review by caalador — APPROVED** [review]

**Pull request merged**